### ISSUE TITLE
bug(profile): firebase v5.0.* causes userData.uid is undefined when creating profile

### ISSUE THREAD

**Issue by nasimoyz** [prescottprue/react-redux-firebase#468]
**What is the current behavior?**

I'm probably missing something obvious, but I keep getting an error where `undefined` is provided in place of string when trying to write profile with creating new user.

![image](https://user-images.githubusercontent.com/1500565/40097639-4073e25e-58a5-11e8-8555-4b5ca079b56a.png)

It appears that in `createUserProfile` it attempts to access `uid` on `userData`, but it actually is found a layer deeper on `user`.

![image](https://user-images.githubusercontent.com/1500565/40097676-6fac5704-58a5-11e8-8d79-4d967f0a24b6.png)

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via [codesandbox](https://codesandbox.io/) or similar.**

onSubmit:
```
    this.props.firebase.createUser({
      email: values.email,
      password: values.password,
    }, {
      email: values.email,
      organization: values.organization,
    });
```

rrfConfig:
```
const rrfConfig = {
  enableLogging: false,
  useFirestoreForProfile: true,
  userProfile: 'users',
};
```

**Which versions of dependencies, and which browser and OS are affected by this issue? Did this work in previous versions or setups?**

I'm using:
```
"firebase": "^5.0.1",
"react-redux-firebase": "^2.1.1",
"redux-firestore": "^0.5.2",
```

**Comment by kx-emeka** [comment]
I should mention that this issue also affects firebase.

Looks like an issue with compatibility with firebase versions 5.X.X

Downgrading to a 4.X.X version (4.6.0 to be precise) resolved this for me. Will continue to follow this and upgrade firebase back to 5.X.X when it is closed.

**Comment by prescottprue** [comment]
@kx-emeka Great to know, thanks for reaching out.

Looking into this today.

**Comment by prescottprue** [comment]
Fixed in [`v2.1.2` release](https://github.com/prescottprue/react-redux-firebase/releases/tag/v2.1.2).
